### DISCUSSION TITLE
Possible to use github enterprise with projen

### DISCUSSION THREAD

**Discussion by pharindoko** [projen/projen#2222] [Q&A] [answered]
Hey,

is it possible to use GH Enterprise with Projen ?

My assumptions:
1. We would need to have an option to set the Github Url in config.
2. Would mean that the gh actions workflows needs to have self-hosted label at least for upgrade-main workflow...

```yml
runs-on: [self-hosted]
```

br,

flo

**Comment by mrgrain** [comment, marked as answer]
Hi @pharindoko 

> * We would need to have an option to set the Github Url in config.

Not exactly sure what you mean by this.

> * Would mean that the gh actions workflows needs to have self-hosted label at least for upgrade-main workflow...

`runsOn: ["self-hosted"]` should be supported by all worflows. If there's one missing please open an issue for it.

----

PS: Converted this to a discussion as the requirements and/or existing features appear to be unclear. You are very welcome to open issues once we have identified any bugs or specific feature gaps!

**Reply by pharindoko** [reply]
Just wanted to know that before I try it out in our gh enterprise environment.
Thanks for your reply.

**Comment by duckpuppy** [comment]
Adding to this...
There are places in the code that assume a `github.com` url. I am looking at submitting a PR that would address these things, but I'm not sure what the "projen way" of doing this would be.

Specifically [here](https://github.com/projen/projen/blob/cc36f7d958a87ef4df058f45dc3f1c38f69a4931/src/github/workflow-actions.ts#L11) and [here](https://github.com/projen/projen/blob/cc36f7d958a87ef4df058f45dc3f1c38f69a4931/src/cli/macros.ts#L21C27-L21C27) assume `github.com` as the URL.

**Reply by mrgrain** [reply]
Thanks for looking into this!
[workflow-actions.ts](https://github.com/projen/projen/blob/cc36f7d958a87ef4df058f45dc3f1c38f69a4931/src/github/workflow-actions.ts#L11) - This is probably going to be an option that needs to be passed through
[macros.ts](https://github.com/projen/projen/blob/cc36f7d958a87ef4df058f45dc3f1c38f69a4931/src/cli/macros.ts#L21C27-L21C27) - This one is more difficult. It's only used as a [default value if not provided](https://github.com/projen/projen/blob/cc36f7d958a87ef4df058f45dc3f1c38f69a4931/src/cdk/jsii-project.ts#L31) explicitly. There's already a check to use an existing repo. Not sure if there's a way to auto-detect an enterprise instance.

**Reply by duckpuppy** [reply]
I think the fix in `workflow-actions.ts` is simple - I've got a little more testing to do on the GHE side of things (you'd be surprised at the feature disparity between github.com and GHE). It might literally be as simple as this:

```
const SERVER_URL = context("github.server_url");
const RUN_URL = `${SERVER_URL}/${REPO}/actions/runs/${RUN_ID}`;
```

I need to confirm when that context variable was introduced, and make sure it's not a very recent thing.

**Reply by mrgrain** [reply]
Ah yes that would be a good and easy fix!
